src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,158 @@
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create a connection to the DB" ,e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM " + TABLE_NAME;
+        try (Connection connection = ConnectionUtil.getConnection();
+             Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery(sql);
+            List<Book> allBooks = new ArrayList<>();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject(FIRST_INDEX_SQL_DATA, Long.class);
+                String title = resultSet.getString(TITLE_NAME);
+                BigDecimal price = resultSet.getObject(PRICE_NAME, BigDecimal.class);
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                allBooks.add(book);
+            }
+            return allBooks;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create a connection to the DB" ,e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE " + TABLE_NAME + " SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            validateTableConnection(connection, TABLE_NAME);
+            statement.setString(FIRST_INDEX_SQL_DATA, book.getTitle());
+            statement.setBigDecimal(SECOND_INDEX_SQL_DATA, book.getPrice());
+            statement.setLong(THIRD_INDEX_SQL_DATA, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < ROW_EXIST_NUM) {
+                throw new DataProcessingException("Expected to insert at leas one row, but inserted 0 row.");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update new book: " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM " + TABLE_NAME + " WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            validateTableConnection(connection, TABLE_NAME);
+            statement.setLong(FIRST_INDEX_SQL_DATA, id);
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < ROW_EXIST_NUM) {
+                throw new DataProcessingException("Expected to insert at leas one row, but inserted 0 row.");
+            }
+            return affectedRows >= 1;

```

Comment:
More correct to do like this.
```suggestion
            return affectedRows > 0;

```